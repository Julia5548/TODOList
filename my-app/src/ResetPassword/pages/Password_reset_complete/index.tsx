import React from 'react'
import { makeStyles, Typography } from '@material-ui/core'
import { NavLink } from 'react-router-dom'

const useStyles = makeStyles((theme) => ({

    page: {
        marginTop: theme.spacing(8),
        display : 'flex',
        flexDirection: 'column',
        alignItems : 'center',
    },
}))

export const Password_Reset_Complete : React.FC = () => {
    const classes = useStyles()
    return(
            <div className = {classes.page}>
                <Typography variant = 'h5' component = "h1">
                    Все ок!
                </Typography>
                <NavLink to="/" >
                        Авторизация
                </NavLink>
            </div>
    )
}

export default Password_Reset_Complete;