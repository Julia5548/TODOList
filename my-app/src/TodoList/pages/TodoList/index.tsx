import React, { useEffect, useState } from 'react';
import { reduxForm, InjectedFormProps, Field } from 'redux-form';
import { makeStyles, Typography, TextField, Button, AppBar, Toolbar, Grid} from '@material-ui/core';
import { RouteComponentProps } from 'react-router';
import { ITodo } from '../../../interface';
import ListTask from '../../components/ListTask';
import { withRouter } from 'react-router-dom';

const renderTextField = ({
    input, 
    label,
    meta : {touched, error, invalid},
    ...custom
 }) => (
    <TextField
        autoComplete = {label}
        label = "Наименование задачи"
        margin = 'normal'
        fullWidth
        id = 'text_field'
        name='task'
        error={touched && invalid}
        helperText={touched && error}
        required
        type = {label}
        variant = 'standard'
        {...input}
        {...custom}
    />
)

export function validations(values){
    const error = {};
    if(!values['task']){
        error['task'] = 'Необходимо ввести наименование задачи'
    }
    return error
}

const useStyles = makeStyles((theme) => ({

    root : {
        flexGrow : 1,
    },
    title : {
        flexGrow: 1,
    },
    submit : {
        margin: theme.spacing(3.5, 0, 0, 2),
        textAlign: 'center',
    },
    textField :{
        textAlign: 'center',
        marginLeft: theme.spacing(3),
        color: theme.palette.text.secondary,
    },
}))


export const WindowTask : React.FC<ITodo & InjectedFormProps<{}, ITodo> & RouteComponentProps<{pk: string}>> = ({
   match: { params },
   history, 
   ...props
}: any) => {
    const classes = useStyles();
    const { pk } = params;
    const[todoList, setTodo] = useState<ITodo[]>([])
    const [response, setResponse] = useState([''])    

    const handleSubmit = ( values : any ) => {

        const newTask : ITodo = {
            id : null,
            user : pk,
            name : values.task,
            completed : false
        }

        props.onAddTask(newTask)
        setResponse([newTask.name])
        console.log('RESPONSE_CREATE : ', response)
    }

    const onToggle = (task : ITodo) =>{
        props.onToggle(task)
        setResponse([task.completed + ' ' + task.name])
        console.log('RESPONSE_TOGGLE : ', response)
    }

    const onRemove = (task : ITodo) =>{
        props.onRemove(task)
        setResponse(['remove_task ' + task.name])
        console.log('RESPONSE_REMOVE : ', response)
    }

    const handleLogout = () => {
        localStorage.removeItem('token')
        props.onLogout()
        history.push('/')
    }


    useEffect(() => {
        if(localStorage.getItem('token')){
            console.log('FETCHING...')
            fetch('http://127.0.0.1:8000/api/task_list/' + pk,
            {
                mode: 'cors',
                headers: {
                    Authorization : 'JWT ' + localStorage.getItem('token')
                },
            })
            .then(response => response.json())
            .then(data => {
                setTodo(data)
                //console.log(data)
        
            })
            .catch((error) => console.log('ERROR1: ', error))
        }
    }, [response])


    return(
        <div className = {classes.root}>
            <AppBar position = "static">
                <Toolbar>
                    <Typography variant="h6" className = {classes.title}>
                        {props.username}
                    </Typography>
                    <Button color="inherit" onClick = {() => handleLogout()}>
                        Выход
                    </Button>
                </Toolbar>
            </AppBar>
            <form onSubmit = {props.handleSubmit(handleSubmit)}>
                    <Grid container
                        spacing={2}
                        direction="row"
                        justify="center"
                        alignItems="flex-start">
                        <Grid item xs >
                            <Field name = "task" 
                            component = {renderTextField} 
                            id= 'field_task'
                            label = "string"
                            className = {classes.textField}/>
                        </Grid>
                        <Grid item xs={3}>
                            <Button 
                                name= "create_task"
                                type="submit"
                                variant = "contained"
                                color = "primary"
                                className = {classes.submit}>
                                 Создать
                            </Button>
                        </Grid>
                    </Grid>    
            </form>
            <ListTask todoList = { todoList } onRemove = {onRemove} onToggle = {onToggle} />
           
        </div>
    )
}

const form = reduxForm<{}, ITodo>({
    form : 'createTask', 
    validate: validations
})(withRouter(WindowTask));

export default form