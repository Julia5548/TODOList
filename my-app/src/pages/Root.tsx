import React from 'react';
import { Route,Switch, Redirect } from 'react-router-dom';
import  SignIn  from './authorization/sign_in';
import  SignUp  from './authorization/sing_up/index';
import  ResetPassword  from './authorization/reset_password/index';
import DisplayTodo from './todo-list/display-todo';
import { connect } from 'react-redux';


const mapStateToProps = (state) => ({
    isLoggedIn : state.user_data.isLoggedIn
})

interface IProps{
    isLoggedIn : boolean;
}

export const Root = (props: IProps) => {
    console.log(props)
    return(
        <Switch>
            <Route exact path = "/">
                <SignIn />
            </Route>
            <Route path = "/registration">
                <SignUp />
            </Route>
            <Route path = "/reset">
                <ResetPassword />
            </Route>
            <PrivateRoute path = "/todo/:pk" isLoggedIn = {props.isLoggedIn}>
                <DisplayTodo />
            </PrivateRoute> 
        </Switch>
);
}



function PrivateRoute({children, isLoggedIn,  ...rest}) {
    return(
        <Route
            {...rest}
            render= {({ location }) => 
                isLoggedIn ? 
                (
                    children
                ):(
                    <Redirect 
                        to= {{
                            pathname: '/',
                            state : {from : location}
                        }}
                    />
                )
            }
        />
    );
}

export default connect(mapStateToProps)(Root);
