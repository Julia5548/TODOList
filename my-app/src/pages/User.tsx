import React, { useEffect } from 'react';
import { connect, useSelector } from 'react-redux';
import Route from '../routes';
import { RootState } from '../store/reducers';
import {  onCurrentUserAction, onLogoutAction } from '../store/actions';
import { onGetTokenAction , onGetUserAction} from '../store/actions';
import { IUser } from '../interfaces/IUser';


const mapDispatchToProps = (dispatch) => {
    return({
        
        onGetToken : () => { 
            dispatch(onGetTokenAction()) 
        },

        onCurrentUser : (current_user: IUser) => { 
            dispatch(onCurrentUserAction(current_user))
        },
        onGetUser : () => {
            dispatch(onGetUserAction())
        },
        onLogout : () => { 
            dispatch(onLogoutAction()) 
        }
    })
}

const mapStateToProps = (state) => ({
    username : state.user_data.username,
    isErrorAuth : state.user_data.isErrorAuth
})

const User = (props : any) => {

    if (localStorage.getItem('token')){
        props.onGetToken();
    }

    const current_state_user = useSelector((state : RootState) => state.user_data.isLoggedIn);

    useEffect(() => {
        if(current_state_user){
            try{
                fetch('http://127.0.0.1:8000/api_users/current_user/',
                    {
                        mode: 'cors',
                        method : 'GET',
                        headers: {
                            Authorization : 'JWT ' + localStorage.getItem('token')
                        }
                    }
                )
                .then(response => 
                    response.json()
                )
                .then(data => {
                    const current_user : IUser = {
                        id : data.id, 
                        username : data.username, 
                        isLoggedIn : true,
                        isErrorAuth : false
                    };
                    if(current_user.id !== undefined){
                        props.onCurrentUser(current_user);
                    }else{
                        props.onLogout();
                    }
                })
            }catch(error){
                console.log('ERROR: ', error)
            }
        }
    }, [current_state_user])

    return (
        <Route {...props}/>
    );
}

export default connect(mapStateToProps, mapDispatchToProps)(User);