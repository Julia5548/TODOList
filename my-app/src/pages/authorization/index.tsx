import React from 'react';
import  SignUp  from '../authorization/sing_up/index';
import  ResetPassword  from '../authorization/reset_password/index';
import { Switch, Route } from 'react-router-dom';
import DisplayTodo from '../todo-list/display-task';


const Router = ( props : any) => {
    return(
        <Switch>
            <Route path = "/registration">
                <SignUp />
            </Route>
            <Route path = "/reset">
                <ResetPassword />
            </Route>
            <Route path = "/todo/:pk">
                <DisplayTodo />
            </Route> 
        </Switch>
    );
}

export default Router