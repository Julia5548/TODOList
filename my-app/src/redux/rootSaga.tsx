import { all } from 'redux-saga/effects'
import { watchCreateTask, watchRemoveTask, watchToggleTask, watch_login_user, watch_create_user, watch_reset_password } from '../ pages/User';


export function* rootSaga(){
    yield all ([
        watchCreateTask(),
        watchToggleTask(),
        watchRemoveTask(),
        watch_create_user(),
        watch_login_user(),
        watch_reset_password()
    ])
}