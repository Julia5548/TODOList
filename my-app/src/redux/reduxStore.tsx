import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { reducer as formReducer } from 'redux-form';
import { reducer as modal } from 'redux-modal';
import createSagaMiddleware from 'redux-saga';
import {rootSaga} from '../redux/rootSaga';
import {user_reducer} from '../ pages/User'


const reducer = combineReducers({ 
    user_data : user_reducer,
    form: formReducer,
    modal
})

const sagaMiddleware = createSagaMiddleware()
const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION__COMPROSE || compose
const store = createStore(reducer, composeWithDevTools(
    applyMiddleware(sagaMiddleware),
    composeEnhancers()
));

sagaMiddleware.run(rootSaga)

export type RootState  = ReturnType<typeof reducer>
export default store