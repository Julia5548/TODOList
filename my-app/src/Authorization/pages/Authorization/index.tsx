import React from 'react'
import { reduxForm, InjectedFormProps, Field } from 'redux-form'
import { makeStyles, Typography, TextField, Button, Grid } from '@material-ui/core'
import { NavLink, useHistory, useRouteMatch } from 'react-router-dom';
import { IUser } from '../../../interface';
import { connect, useSelector } from 'react-redux';
import { RootState } from '../../../redux/reduxStore';


interface IProps{
    name : string;
    password : string;
}


const renderTextField = 
({  input, 
    label,
    meta: { touched, error, invalid }, 
    ...custom 
}) => (
    <TextField
        autoComplete = {label}
        fullWidth
        label = {label}
        margin = 'normal'
        name={label}
        error={touched && invalid}
        helperText={touched && error}
        required
        type = {label}
        variant = 'standard'
        {...input}
        {...custom}
    />
)

const useStyles = makeStyles((theme) => ({

    page: {
        marginTop: theme.spacing(8),
        display : 'flex',
        flexDirection: 'column',
        alignItems : 'center',
    },
    submit : {
        margin: theme.spacing(3, 0, 2),
    },
}))

export const Authorization : React.FC = (props : any) => {

    const classes = useStyles()
    const history = useHistory()

    const id_user = useSelector((state : RootState) => state.user_data.id)

    const handleSubmit = (values: IProps) => {
        const user : IUser = {
            username : values.name,
            password : values.password,
            id : 0,
            logged_in : false
        }

        props.onLoginUser(user, history)
        
    }
    
    return (
            <div className = {classes.page}>
                <Typography variant = 'h5' component = "h1">
                    Авторизация
                </Typography>
                <form onSubmit={props.handleSubmit(handleSubmit)}>
                    <Field name="name" component={renderTextField} label="name" />
                    <Field name="password" component={renderTextField} label="password" />
                    <Button
                        type="submit"
                        fullWidth
                        variant = "contained"
                        color = "primary"
                        className = {classes.submit}>
                         Войти
                    </Button>
                    <Grid container>
                        <Grid item xs>
                            <NavLink to="/reset">
                                Забыли пароль?
                            </NavLink>
                        </Grid>
                        <Grid item>
                            <NavLink to="/registration" >
                                Вы еще не зарегистрированы?<br/>Регистрация
                            </NavLink>
                        </Grid>
                    </Grid>
                </form>
            </div>
    )
}

const form = reduxForm({
    form: 'auth'
})(Authorization);
    
export default form