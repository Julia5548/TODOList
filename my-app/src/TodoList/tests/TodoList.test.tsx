import React from 'react'
import {WindowTask} from '../pages/TodoList'
import ReduxFormWindowTask from '../pages/TodoList'
import {reducer as formReducer} from 'redux-form'
import {shallow, mount, ShallowWrapper} from 'enzyme'
import { Field } from 'redux-form';
import { Button } from '@material-ui/core';
import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

describe('<WindowTask />', () => {
    let wrapper: any;

    const props:any = {
        handleSubmit : jest.fn(),
    }

    it('checks if initial state is equal to 0', () => {
        wrapper = shallow(
            <WindowTask {...props}  match = {{params : {pk : '1'}}}/>
        );

        const taskField = wrapper.find(Field);
        // console.log(toJson(taskField))
        expect(taskField.prop('name')).toBe('task');
        expect(taskField.prop('label')).toBe('string');
    });

    it('button create', () => {
        const create_button = wrapper.find(Button).find({name : 'create_task'});
    
        expect(create_button.prop('type')).toEqual('submit');
        expect(create_button.prop('children')).toEqual('Создать');
        // console.log(toJson(create_button))
    });
})

describe('Redux-Form WindowTask', () =>{
    let store : any;
    let wrapper : any;

    const props:any = {
        handleSubmit : jest.fn(),
    }

    it('displayed correctly',() => {
        store = createStore(combineReducers({form : formReducer}))
        wrapper = mount(
            <Provider store= {store}>
                <MemoryRouter initialEntries= {['todo/1']}>
                    <ReduxFormWindowTask {...props} match = {{params : {pk : '1'}}}/>
                </MemoryRouter>
            </Provider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})