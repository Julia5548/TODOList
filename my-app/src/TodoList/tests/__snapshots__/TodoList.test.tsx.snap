// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WindowTask /> defines createTask form fields renders task field 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "qn096e",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        Object {
          "hash": "",
          "key": "1prqbl",
          "pathname": "/page/10",
          "search": "",
          "state": undefined,
        },
        Object {
          "hash": "",
          "key": "5ort8p",
          "pathname": "/next",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 1,
      "length": 3,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "1prqbl",
        "pathname": "/page/10",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    path="/page/:page"
  >
    <WindowTask
      handleSubmit={[MockFunction]}
    />
  </Route>
</Router>
`;

exports[`Redux-Form WindowTask displa correctly 1`] = `ReactWrapper {}`;

exports[`Redux-Form WindowTask displayed correctly 1`] = `ReactWrapper {}`;

exports[`Redux-Form WindowTask render correctly 1`] = `ReactWrapper {}`;

exports[`Redux-Form WindowTask shows error when name task is set to blank 1`] = `ReactWrapper {}`;
